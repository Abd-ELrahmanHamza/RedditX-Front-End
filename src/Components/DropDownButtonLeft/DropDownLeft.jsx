// imports
import React, { useEffect, useRef, useState } from "react";
import Dropdown from "react-bootstrap/Dropdown";
import { DropDownLeftStyled } from "./DropDownLeft.styled";
import { AiFillHome } from "react-icons/ai";
import DropDownMenu from "Components/DropDownMenu/DropDownMenu";
import CustomToggle from "Components/CustomToggle/CustomToggle";
import SideBar from "Components/SideBar/SideBar";
import { GrFormDown } from "react-icons/gr";

/**
 * Component that has all your accessibility and profile in drop-down list.
 *
 * @returns {React.Component}
 */

const DropDownLeft = () => {
  const dropdownMenuRef = useRef(null);
  const toggleDropdownRef = useRef(null);
  const dropDownItemRef = useRef(null);
  const [show, setShow] = useState(false);

  const handleToggle = () => {
    if (!show && showSideBar) {
      console.log("hereee");
      return;
    }
    console.log("baraa");
    setShow(!show);
  };
  useEffect(() => {
    const handleClickOutside = (event) => {
      if (
        dropdownMenuRef.current &&
        !dropdownMenuRef.current.contains(event.target) &&
        toggleDropdownRef.current &&
        !toggleDropdownRef.current.contains(event.target)
      ) {
        setShow(false);
      }
    };
    // Bind the event listener
    document.addEventListener("click", handleClickOutside);
    return () => {
      // Unbind the event listener on clean up
      document.removeEventListener("click", handleClickOutside);
    };
  }, [dropdownMenuRef, toggleDropdownRef, dropDownItemRef]);

  const [showSideBar, setShowSideBar] = useState(false);
  return (
    <DropDownLeftStyled>
      <Dropdown.Toggle
        as={CustomToggle}
        id="dropdown-home"
        onClick={handleToggle}
        ref={toggleDropdownRef}
      >
        <div className={"all-drop"}>
          <div className={"home-ico"}>
            <span>
              <AiFillHome />
            </span>
            <span>
              <h6>Home</h6>
            </span>
          </div>
          <SideBar
            showButton={show}
            showSideBar={showSideBar}
            setShowSideBar={setShowSideBar}
          />
        </div>
        <span className={"down-form"}>
          <GrFormDown />
        </span>
      </Dropdown.Toggle>
      {
        <DropDownMenu
          dropDownItemRef={dropDownItemRef}
          dropdownMenuRef={dropdownMenuRef}
          show={show}
        />
      }
    </DropDownLeftStyled>
  );
};
export default DropDownLeft;
