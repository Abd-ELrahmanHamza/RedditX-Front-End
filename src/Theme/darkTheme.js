/**
 * dark theme colors
 *
 * @type {{id:string}}
 */
const darkTheme = {
  // The id of the theme
  id: "dark",
  // The canvas color is the background color of the website
  canvas: "#030303",
  // The colors of the text on the website
  color: {
    // The primary color of the text on the website
    primary: "#d7dadc",
    // The color of the muted text on the website
    muted: "#7c7c7c",
    // The color of the links on the website
    link: "#0079d3",
  },
  // The colors of the backgrounds in the website
  background: {
    primary: "#1a1a1b",
  },
  // The colors of the lines in the website
  lineColor: {
    primary: "#343536",
  },
};

export default darkTheme;
