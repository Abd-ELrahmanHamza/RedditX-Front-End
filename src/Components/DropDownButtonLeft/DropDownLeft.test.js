import React from "react";
import { BrowserRouter, Routes } from "react-router-dom/dist";
import darkTheme from "Theme/darkTheme";
import { ThemeProvider } from "styled-components";
import { screen, render, fireEvent, waitFor } from "@testing-library/react";
import { shallow } from "enzyme";
// Import contexts
import {
  AuthProvider,
  useAuth,
} from "Features/Authentication/Contexts/Authentication";

// import Components
import DropDownLeft from "./DropDownLeft";

jest.mock("Features/Authentication/Contexts/Authentication", () => ({
  __esModule: true, // this property makes it work
  useAuth: function () {
    return {
      getToken: jest.fn().mockReturnValue("token"),
    };
  },
  AuthProvider: ({ children }) => {
    return (
      <mock-authprovider data-testid="auth-provider">
        {children}
      </mock-authprovider>
    );
  },
}));

describe("Drop Down Left Component", () => {
  shallow(<DropDownLeft />);
  test("Drop Down Left should be appeard", async () => {
    render(
      <BrowserRouter>
        <AuthProvider>
          <ThemeProvider theme={darkTheme}>
            <DropDownLeft />
          </ThemeProvider>
        </AuthProvider>
      </BrowserRouter>
    );
    expect(screen.queryByTestId("DropDownLeft")).not.toBeInTheDocument();
  });
});
