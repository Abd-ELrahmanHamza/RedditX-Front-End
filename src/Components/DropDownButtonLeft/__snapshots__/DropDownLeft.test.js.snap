// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drop Down list Component this is a test for Drop Down list Component 1`] = `
<Styled(Dropdown)>
  <DropdownToggle
    as={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      }
    }
    id="dropdown-home"
    onClick={[Function]}
  >
    <styled.div>
      <styled.div>
        <span>
          <AiFillHome />
        </span>
        <span>
          <styled.h6>
            Home
          </styled.h6>
        </span>
      </styled.div>
    </styled.div>
    <div>
      <SideBar
        setShowSideBar={[Function]}
        showButton={false}
        showSideBar={false}
      />
      <styled.span>
        <GrFormDown />
      </styled.span>
    </div>
  </DropdownToggle>
  <DropDownMenu
    dropdownMenuRef={
      Object {
        "current": null,
      }
    }
    show={false}
  />
</Styled(Dropdown)>
`;
